core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -154,8 +154,16 @@ public void start() throws IOException, IllegalStateException {
 
                     frameMat.copyTo(frameOutputSocket.getValue().get());
                     frameOutputSocket.setValue(frameOutputSocket.getValue().get());
-                    long thisMoment = System.currentTimeMillis();
-                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
+
+                    final long thisMoment = System.currentTimeMillis();
+                    final long elapsedTime = thisMoment - lastFrame;
+                    if (elapsedTime == 0) {
+                        // This can actually happen: https://github.com/WPIRoboticsProjects/GRIP/issues/199
+                        frameRateOutputSocket.setValue(Double.NaN);
```

Comment:
Actually, I suggested that if the frame time is zero, then check the source and see why that is the case and how it can happen.

-- 
Brad Miller


On December 10, 2015 at 2:01:03 PM, Thomas Clark (notifications@github.com) wrote:

In core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java:

> @@ -154,8 +154,16 @@ public void start() throws IOException, IllegalStateException {
>   
>                      frameMat.copyTo(frameOutputSocket.getValue().get());
>                      frameOutputSocket.setValue(frameOutputSocket.getValue().get());
> -                    long thisMoment = System.currentTimeMillis();
> -                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
> +
> +                    final long thisMoment = System.currentTimeMillis();
> +                    final long elapsedTime = thisMoment - lastFrame;
> +                    if (elapsedTime == 0) {
> +                        // This can actually happen: https://github.com/WPIRoboticsProjects/GRIP/issues/199
> +                        frameRateOutputSocket.setValue(Double.NaN);
If the frame rate is zero? Or the elapsed time?

—
Reply to this email directly or view it on GitHub.